### ISSUE TITLE
[package] lcms/2.14: `ConanException: You must provide a valid SDK path for cross-compilation.`

### ISSUE THREAD

**Issue by DominikDeak** [conan-io/conan-center-index#15339]
### Description

When trying to cross compile `lcms` library on Apple M1 with `x86_64` architecture target, I get the following error:

```
ERROR: lcms/2.14: Error in generate() method, line 56
	tc = MesonToolchain(self)
	ConanException: You must provide a valid SDK path for cross-compilation.
```

Invocation command:

`conan install "lcms/2.14@" --update --build outdated --profile:h default --profile:b default --settings:h arch=x86_64 -o lcms:shared=True`

### Package and Environment Details

* Package Name/Version: **lcms/2.14**
* Operating System+version: **Apple M1 macOS Ventura 13.1 (22C65)**
* Compiler+version: **apple-clang 14.0**
* Docker image: **n/a**
* Conan version: **Conan version 1.56.0**
* Python version: **Python 3.10.9**

### Conan profile

[settings]
os=Macos
os.version=11.0
os_build=Macos
arch=armv8
arch_build=armv8
compiler=apple-clang
compiler.version=14.0
compiler.libcxx=libc++
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

`conan install "lcms/2.14@" --update --build outdated --profile:h default --profile:b default --settings:h arch=x86_64 -o lcms:shared=True`

### Logs

<details><summary>Click to expand log</summary>

```
> conan install "lcms/2.14@" --update --build outdated --profile:h default --profile:b default --settings:h arch=x86_64 -o lcms:shared=True 
Configuration (profile_host):
[settings]
arch=x86_64
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=14.0
os=Macos
os.version=11.0
os_build=Macos
[options]
lcms:shared=True
[build_requires]
[env]

Configuration (profile_build):
[settings]
arch=armv8
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=14.0
os=Macos
os.version=11.0
os_build=Macos
[options]
[build_requires]
[env]

lcms/2.14: WARN: Package binary is corrupted, removing: 22a3e8ce56a87b0216515b2f6821f5b98ddecd3f
ninja/1.11.1: WARN: Can't update, no package in remote
ninja/1.11.1: Package is up to date
meson/1.0.0: Package is up to date
Installing package: lcms/2.14
Requirements
    lcms/2.14 from 'conancenter' - Cache
Packages
    lcms/2.14:22a3e8ce56a87b0216515b2f6821f5b98ddecd3f - Build
Build requirements
    meson/1.0.0 from 'conancenter' - Cache
    ninja/1.11.1 from 'conancenter' - Cache
Build requirements packages
    meson/1.0.0:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    ninja/1.11.1:1affe81c96684c6628cfec7b81236601b9e78183 - Cache

Cross-build from 'Macos:armv8' to 'Macos:x86_64'
Installing (downloading, building) binaries...
ninja/1.11.1: Already installed!
meson/1.0.0: Already installed!
lcms/2.14: Applying build-requirement: meson/1.0.0
lcms/2.14: Applying build-requirement: ninja/1.11.1
lcms/2.14: WARN: Build folder is dirty, removing it: /Users/dom/.conan/data/lcms/2.14/_/_/build/22a3e8ce56a87b0216515b2f6821f5b98ddecd3f
lcms/2.14: Copying sources to build folder
lcms/2.14: Building your package in /Users/dom/.conan/data/lcms/2.14/_/_/build/22a3e8ce56a87b0216515b2f6821f5b98ddecd3f
lcms/2.14: Generator txt created conanbuildinfo.txt
lcms/2.14: Calling generate()
ERROR: lcms/2.14: Error in generate() method, line 56
	tc = MesonToolchain(self)
	ConanException: You must provide a valid SDK path for cross-compilation.

```

</details>

**Comment by SpaceIm** [comment]
It's not a conan-center bug. It comes from `MesonToolchain` which needs this information. In the meantime I think it could find this information itself, so you can submit an issue in conan client repo.

**Comment by DominikDeak** [comment]
@SpaceIm thanks for looking into this and providing work-around.

**Comment by SpaceIm** [comment]
https://github.com/conan-io/conan/pull/12923 has been merged in conan client, so from conan 1.58.0 (not yet released) you won't have to explicitly set `tools.apple:sdk_path` when cross-building meson based recipes.
